### PR TITLE
Release v0.3.0 mixed-grid Continuum handoff, seam bridge, and decode context

### PR DESCRIPTION
**Pull request by xmarre**
## Summary

This PR finalizes **MiniMax H3 Flow-Aligned Regenerate v0.3.0**.

The main release result is the validated exact-prefix Continuum path:

- **Progressive Mixed-Grid Continuum [Experimental]** keeps the authoritative target-grid protected prefix for H3 conditioning while sampling a genuine lower-resolution generated suffix;
- it runs the exact handoff probe under the same mixed topology;
- it performs the learned 3D latent transfer before a fresh target-grid refinement stage;
- it composes with VDN-H3 external-sequence API 2, Spectrum/SA-PECE, DiffAid and Untwisting RoPE;
- the one-token `suffix_dc_bridge` fixes the previously observed Continuum boundary flash in the matched production workflow, including multiple chunk boundaries.

The generic **Progressive Handoff (Target Input)** remains a generic node and does **not** expose or apply the Continuum seam bridge.

## Mixed-Grid exact-prefix contract

For exact Native Masked continuation:

1. Snapshot the original model-domain target-grid prefix and target-grid sampler noise.
2. Run the private early sampler on a regular low-grid carrier.
3. Before H3 transformer block 0, independently patchify/project the authoritative target-grid protected prefix using native masked-conditioning semantics and replace only the carrier prefix rows.
4. Keep genuine source-grid suffix rows, mixed RoPE and coherent per-row modulation labels through every transformer block.
5. Before native output projection, discard target-prefix transformer outputs and restore the regular low-carrier layout.
6. Run the exact handoff probe through the same mixed contract in a separate sampler lifetime.
7. Feed the complete low-grid clean sequence to the versioned `learned_3d` provider.
8. Discard the learned prefix output, restore the authoritative target-grid prefix, reconstruct the high-stage conditional state/noise, and start a fresh full-grid sampler lifetime.
9. Require the first high-stage call to be an actual H3 evaluation and verify final prefix exactness.

The authoritative target-grid prefix is never spatially resized for H3 transformer conditioning.

## Suffix DC bridge

Mixed-Grid exposes `suffix_dc_bridge`, default **on**.

The learned 3D transfer produces `[U_prefix | U_suffix]` before its prefix is discarded. Flow computes the per-batch/per-channel spatial-mean difference between `P_exact[-1]` and `U_prefix[-1]`, then transfers that relation onto **only the first generated suffix latent token**.

The bridge is deliberately narrow:

- one generated suffix token;
- fixed weight `1.0`;
- per-channel DC/spatial-mean correction;
- authoritative prefix unchanged;
- later suffix tokens unchanged at bridge application;
- no video-space crossfade;
- no extra H3 transformer NFE.

Matched real-media evidence on RTX Pro 6000 reported approximately:

- uncorrected exact-boundary DC RMS: `0.207763`;
- corrected exact-boundary DC RMS: `0.093093`;
- learned-upscaler native boundary DC RMS: `0.093093`;
- corrected tokens: `1`;
- weight: `1.0`;
- `final_prefix_exact=true`.

The visible boundary flash was removed, and the multi-boundary Continuum run was also clean.

## Target-Sparse result

**Progressive Target-Sparse Continuum [Experimental]** remains available as a research/control path but is not promoted for production quality.

It was already tested with real decoded media. Because it sparsifies the early target-grid hidden stream without performing the learned latent upscale, approximation errors can cascade into later generation. Observed failures included skin imperfections, unstable/odd clothing details and spurious background additions.

That negative result closes the previous quality gate: the release recommendation for exact-prefix progressive Continuum is **Mixed-Grid + learned 3D transfer**, not Target-Sparse.

Target-Sparse still exposes the dedicated Continuum-only one-token bridge at its full-grid high-stage boundary for research/control use. This does not alter the broader quality conclusion above.

## Generic Target Input separation

**Progressive Handoff (Target Input)** is not a Continuum-specific exact-prefix node.

- normal unprotected calls can use the private low/probe/high progressive path;
- exact protected prefixes conservatively fall back to a target-grid sampler lifetime;
- `bicubic` and optional `learned_3d` remain available for the ordinary handoff path;
- it has no `suffix_dc_bridge` UI or runtime behavior.

Regression coverage enforces this separation.

## Continuum Decode Context

Adds **MiniMax H3 Continuum Decode Context** for a separate native temporal-VAE boundary issue.

The node supplies real future latent context to the preceding chunk's decode-only tensor at exact physical joins while preserving accepted sampling latents, continuation state, masks, plan and audio. The unchanged assembly trim discards the extra decode tail.

This is independent from the latent DC bridge; the bridge is what fixed the validated Mixed-Grid flash.

## VDN-H3 interoperability

The source-contract lane is coordinated with the merged VDN-H3 v1.5.0 release commit:

`xmarre/ComfyUI-VDN-H3@1fc9d0d979683e3410587fac17358d3dd583e551`

Mixed-Grid uses API 2 / `vdn_h3_external_sequence_v1` with `topology=mixed_grid_low_suffix`. VDN retains the learned dense softmax gate and disables only geometry-dependent local-window/linear-complement processing while the mixed sequence is active. The fresh target-grid high stage returns to normal VDN execution.

VDN v1.5.0 also contains the final Comfy-managed lifecycle work and backwards-compatible `ApplyVDNH3Advanced` positional-workflow migration.

## Release / README updates

- Package version: **0.3.0**.
- README is rewritten around the actual recommended paths rather than development gates.
- Mixed-Grid docs record the successful bridge/multi-boundary result.
- Target-Sparse docs record the negative decoded-media result instead of asking for another acceptance sweep.
- `RELEASE_NOTES.md` contains the v0.3.0 release notes.
- Existing release automation creates the GitHub ZIP + `SHA256SUMS` from the exact CI-tested main commit.
- Existing publish automation publishes the changed `0.3.0` package to the Comfy Registry only after successful main CI.

## Validation

Release staging tree:

`696c4885939ed8765c62c04bf5f7ca763269b271`

Staging CI:

`33987734118` — **success**

It includes:

- Python 3.10 / 3.11 / 3.12 / 3.13 lanes;
- Ruff and format checks;
- full pytest suite;
- compileall;
- sdist/wheel build;
- Apache-2.0 distribution metadata checks;
- isolated wheel import;
- pinned native/sibling source-contract checks, now against the merged VDN v1.5.0 commit;
- mixed-grid and temporal-VAE native source oracles.

## Final PR topology

- Base: `a0c93a436b703db473287dce90df271ddb32717b`
- Head: `b44547efcd394f2c2b9b14d6ddfcbb2d2950dd86`
- Tree: `696c4885939ed8765c62c04bf5f7ca763269b271`
- **1 commit ahead / 0 behind**

This PR is release-ready once the exact final-head PR CI completes successfully.

### CODE REVIEW

**Comment by xmarre** [comment]
@coderabbitai review

File: h3_flow_regenerate/target_sparse.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    hidden = int(compact_video.shape[-1])
+    anchors = anchors.reshape(plan.target_t, plan.source_patch_h, plan.source_patch_w, hidden)
+    work = anchors.permute(0, 3, 1, 2).to(torch.float32)
+    align_corners = plan.source_patch_h > 1 and plan.source_patch_w > 1
+    lifted = F.interpolate(
+        work,
+        size=(plan.target_patch_h, plan.target_patch_w),
+        mode="bilinear",
+        align_corners=align_corners,
+    )
```
[the remark this answers is not in the document]
Comment (xmarre) [reply h3_flow_regenerate/target_sparse.py]: Fixed. PyTorch bilinear interpolation with `align_corners=True` preserves a singleton axis while endpoint-aligning the non-singleton axis, so the lifter now enables endpoint alignment whenever either source patch axis has multiple anchors (`or`, not `and`). Added a parameterized regression covering both 1×2→1×4 and 2×1→4×1 lifts; the validated pre-squash tree collected 163 tests and passed on Python 3.10–3.13.

**Comment by xmarre** [comment]
Final structural validation after the decode-context addition:

- head: `a2d56330737b5db2291a79dbcb9eb7d15435e399`
- tree: `840ad795f64613ec039c0d0201f8606f364665f4`
- base: `a0c93a436b703db473287dce90df271ddb32717b`
- topology: **1 commit ahead / 0 behind**
- changed files: **27**
- mergeable: **yes**
- draft: **yes**

Exact PR-head CI: `33976010249` — **green**.

Main Python 3.12 suite:
- 207 collected
- 196 passed
- 11 skipped native-source cases

Pinned source-contract/native-oracle lane:
- **35 passed**
- ComfyUI `1af040bf022569d7a890241c8dd79b296cda483f`
- Continuum `bf25353d8bec44afea22c89717c4301ce13c2036`
- VDN PR #2 `cd0083291f62203fcb40d578ae58b4caaa1c5ec7`
- plus pinned Spectrum, DiffAid, RefDelta, Untwist and latent-upscaler sources

The new **MiniMax H3 Continuum Decode Context** node is decode-only and does not alter denoising, accepted chunk latents, continuation state, masks, audio or the assembly plan. At each bit-identical native-phase overlap it appends the next five real generated latent tokens to a temporary decode view of the preceding chunk, then relies on the existing assembly plan to discard the 17 extra decoded frames.

The native temporal-decoder oracle reproduces the independent-chunk discrepancy in the five frames immediately before each join and shows bit-identical equivalence to continuous decoding when right context is supplied, for four-chunk tests with 5/22/39-frame overlaps. Non-exact boundaries fail safe and are reported instead of approximated.

This is a structural/native-window result, not decoded-media proof. The next real GPU/media gate is a sampler-identical A/B with Decode Context bypassed vs enabled. That test should establish whether the user's slight visible boundary flash is actually reduced by correcting the proven VAE context discontinuity. Replacement/TRT VAEs need separate validation if their temporal decode contract differs.

VDN PR #2 requires no change for this decoder-boundary correction. No actionable inline review thread is currently open; the old target-sparse endpoint-alignment thread remains resolved/outdated.

**Comment by xmarre** [comment]
@coderabbitai review

**Comment by xmarre** [comment]
@coderabbitai review

Please review current head `a2d56330737b5db2291a79dbcb9eb7d15435e399`, especially the new exact-prefix Continuum decode-context node and native temporal-decoder oracle.

File: tests/test_packaging.py [hunk 2]
```diff
@@ -60,14 +74,25 @@ def test_progressive_nodes_expose_all_selectable_guidance_controls():
...
 
-    target_schema = H3ProgressiveTargetInputHandoff.INPUT_TYPES()
-    assert target_schema["required"]["handoff_transfer"][0] == ["bicubic", "learned_3d"]
-    assert target_schema["required"]["handoff_transfer"][1]["default"] == "bicubic"
-    assert target_schema["optional"]["learned_upscaler"] == ("H3_LATENT_UPSCALER",)
+    for node in (H3ProgressiveTargetInputHandoff, H3ProgressiveTargetSparseHandoff):
+        target_schema = node.INPUT_TYPES()
+        assert target_schema["required"]["handoff_transfer"][0] == ["bicubic", "learned_3d"]
+        assert target_schema["required"]["handoff_transfer"][1]["default"] == "bicubic"
+        assert target_schema["optional"]["learned_upscaler"] == ("H3_LATENT_UPSCALER",)
```
[the remark this answers is not in the document]
Comment (xmarre) [reply tests/test_packaging.py:83]: This finding is based on the wrong class. `H3ProgressiveTargetSparseHandoff` does not override `INPUT_TYPES()`; it inherits the ordinary Target Input schema, so its contract remains `handoff_transfer=["bicubic", "learned_3d"]` with default `bicubic` and the optional `H3_LATENT_UPSCALER` input. The learned-only override at `target_sparse_node.py:103-111` belongs to the subclass `H3ProgressiveMixedGridHandoff`, which intentionally requires `learned_3d`.

The packaging assertion is therefore correct and changing it would encode the wrong Target-Sparse contract. The staged tree was validated with this assertion unchanged: focused bridge/mixed-grid suite 56 passed (8 native-oracle skips), full Python 3.12 suite 209 passed (11 native-oracle skips), Ruff/format/compile/whitespace checks green. No code change is appropriate for this thread.
